code/game/machinery/computer/buildandrepair.dm

```diff
@@ -109,9 +109,45 @@
 			if(P.tool_behaviour == TOOL_SCREWDRIVER)
 				P.play_tool_sound(src)
 				to_chat(user, "<span class='notice'>You connect the monitor.</span>")
-				var/obj/B = new circuit.build_path (loc, circuit)
-				B.setDir(dir)
-				transfer_fingerprints_to(B)
+
+				var/obj/machinery/new_machine = new circuit.build_path(loc)
```

Comment:
what the fuck is this byondism. what does this do.

what do you mean `new EXPRESSION(args)`. WHY DOES THIS EVEN WORK.

kill it with fire

Reply:
huh?

Machines have to reference their circuits in their component parts, or it will be spewed out when its opened

Reply:
> what the fuck is this byondism. what does this do.
> 
> what do you mean `new EXPRESSION(args)`. WHY DOES THIS EVEN WORK.

Technically it works because `circuit.build_path` alone to byond *is a path* rather than an expression.

Reply:
No. I mean this is a syntactic nightmare. You should not be able to call new (expression). it should have to be in a var because the expression evaluates to a path. I don't like that you can just do new (expression)